### PR TITLE
fix(curriculum): Shopping Cart optional braces

### LINKED ISSUE #56686: js beta shopping cart - step 35 code checking too strict for braces
### Describe the Issue

in the js beta shopping cart project , step 36 does not allow the code below to pass
```
[...addToCartBtns].forEach(
  (btn) => btn.addEventListener("click", (event) => {})
);
```
and instead needs braces to be added in order to allow the code to pass as below:
```
[...addToCartBtns].forEach(
  (btn) => {btn.addEventListener("click", (event) => {})}
);
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-oop-by-building-a-shopping-cart/step-35

### Your code

```
[...addToCartBtns].forEach(
  (btn) => btn.addEventListener("click", (event) => {})
);

```

### Expected behavior

I would expect the code without the extra braces to pass.

### Screenshots

_No response_

### System

N/A

### Additional context

forum post that triggered this https://forum.freecodecamp.org/t/learn-basic-oop-by-building-a-shopping-cart-step-35/716369

### PR DESCRIPTION
**Pull request by Supravisor**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56686

With braces
image

Without braces
image

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-oop-by-building-a-shopping-cart/63f0284532742c1b26c7a052.md [hunk 1]
```diff
@@ -14,25 +14,25 @@ Add your callback function to the `forEach` method. It should take a `btn` param
...
 ```js
-assert.match(code, /\[\s*\.\.\.addToCartBtns\s*\]\s*\.\s*forEach\s*\(\s*(?:\(\s*btn\s*\)|btn)\s*=>\s*\{\s*btn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\1\s*,\s*/);
+assert.match(code, /\[\s*\.\.\.addToCartBtns\s*\]\s*\.\s*forEach\s*\(\s*(?:\(\s*btn\s*\)|btn)\s*=>\s*(?:\{\s*btn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\1\s*,\s*|\s*btn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\2\s*,\s*)/);
 ```
 
 You should add an empty callback function to the event listener. Remember to give it an `event` parameter.
 
 ```js
-assert.match(code, /\[\s*\.\.\.addToCartBtns\s*\]\s*\.\s*forEach\s*\(\s*(?:\(\s*btn\s*\)|btn)\s*=>\s*\{\s*btn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\1\s*,\s*(\(\s*event\s*\)|event)\s*=>\s*\{\s*\}\s*\)\s*;?\s*\}\s*\)/);
+assert.match(code, /\[\s*\.\.\.addToCartBtns\s*\]\s*\.\s*forEach\s*\(\s*(?:\(\s*btn\s*\)|btn)\s*=>\s*(?:\{\s*btn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\1\s*,\s*(\(\s*event\s*\)|event)\s*=>\s*\{\s*\}\s*\)\s*;?\s*\}\s*\)|\s*btn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\3\s*,\s*(\(\s*event\s*\)|event)\s*=>\s*\{\s*\}\s*\)\s*;?\s*\s*\))/);
```
Comment (a2937) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-oop-by-building-a-shopping-cart/63f0284532742c1b26c7a052.md:35]: I don't think the last two any length space characters are necessary. Don't get it me wrong, your code works and it works well. It just is bugging me a little bit.

Comment (a2937) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-oop-by-building-a-shopping-cart/63f0284532742c1b26c7a052.md:35]: I don't think the last two any length space characters are necessary. Don't get it me wrong, your code works and it works well. It just is bugging me a little bit.

**Review by a2937 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-oop-by-building-a-shopping-cart/63f0284532742c1b26c7a052.md (hunk 1 above)
Comment (Supravisor) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-oop-by-building-a-shopping-cart/63f0284532742c1b26c7a052.md]: removed extra space

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

Want to give it another look @a2937? This should be fine to merge with the failing tests.

**Review by a2937 — APPROVED** [review]
Congrats on another successful PR.

**Pull request merged**